### PR TITLE
solara: Add virus_on_network network example

### PR DESCRIPTION
**Pull request by rht**
Waiting until https://github.com/projectmesa/mesa/pull/1767 is merged and we have a release.

### CODE REVIEW

File: examples/virus_on_network/app.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    infected_text = str(number_infected(model))
+
+    return "Resistant/Susceptible Ratio: {}<br>Infected Remaining: {}".format(
+        ratio_text, infected_text
+    )
+
+
+def make_plot(viz):
+    # This is for the case when we want to plot multiple measures in 1 figure.
+    # We could incorporate this into core Mesa.
```
Comment (rht) [inline examples/virus_on_network/app.py:53]: Should this be merged to core Mesa for convenience?

**Comment by rht** [comment]
This is ready to merge.

File: examples/virus_on_network/app.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+from matplotlib.ticker import MaxNLocator
+from mesa.experimental import JupyterViz, make_text
+from virus_on_network.model import State, VirusOnNetwork, number_infected
+
+
+def agent_portrayal(graph):
+    def get_agent(node):
+        return graph.nodes[node]["agent"][0]
+
+    width = []  # edge width
```
Comment (Corvince) [inline examples/virus_on_network/app.py:14]: Maybe just call it edge_width in line with edge_color, so it doesn't need the comment?
```suggestion
    edge_width = []
```

File: examples/virus_on_network/app.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    edge_color = []
+    for u, v in graph.edges():
```
Comment (Corvince) [inline examples/virus_on_network/app.py:16]: u, v, w and ec are quite not-descriptive. Maybe call it edge_width, edge_color here and above edge_colors*s* and edge_width*s* ?

File: examples/virus_on_network/app.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+def make_plot(viz):
+    # This is for the case when we want to plot multiple measures in 1 figure.
+    # We could incorporate this into core Mesa.
+    fig = Figure()
+    ax = fig.subplots()
+    measures = ["Infected", "Susceptible", "Resistant"]
+    colors = ["tab:red", "tab:green", "tab:gray"]
+    for i, m in enumerate(measures):
+        color = colors[i]
+        ax.plot(viz.df.loc[:, m], label=m, color=color)
```
Comment (Corvince) [inline examples/virus_on_network/app.py:58]: This needs to be changed to be dependent on model, not viz

**Review by Corvince — COMMENTED** [review]
You need to change from viz to model, otherwise just some minor suggestions.

File: examples/virus_on_network/app.py (hunk 2 above)
Comment (rht) [reply examples/virus_on_network/app.py]: Fixed.

File: examples/virus_on_network/app.py (hunk 4 above)
Comment (rht) [reply examples/virus_on_network/app.py]: Fixed.

File: examples/virus_on_network/app.py (hunk 3 above)
Comment (rht) [reply examples/virus_on_network/app.py:16]: `edge_color` in NetworkX is a list of edge colors. As much as this is confusing, I have to stay consistent with NetworkX. Ditto with the plural for `edge_width`.

`u`, `v` should be fine for undirected graph, just like `i`, `j` for 2D grid.

Comment (rht) [reply examples/virus_on_network/app.py:16]: See `edge_color` in https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw_networkx.html, which may be a scalar or vector.

**Pull request merged**